pennylane/resource/specs.py

```diff
@@ -117,31 +135,244 @@ def _specs_qjit(qjit, level, compute_depth, *args, **kwargs) -> CircuitSpecs:  #
+        original_qnode = qjit.original_function
+    else:
+        raise ValueError(
+            "qml.specs can only be applied to a QNode or qjit'd QNode, instead got:",
+            qjit.original_function,
+        )
+
+    device = original_qnode.device
+
+    if level == "device":
+        resources = _specs_qjit_device_level_tracking(
+            qjit, original_qnode, pass_pipeline_wrapped, compute_depth, *args, **kwargs
+        )
+
+    elif isinstance(level, (int, tuple, list, range, str)):
+        if compute_depth:
+            warnings.warn(
+                "Cannot calculate circuit depth for intermediate transformations or compilation passes."
+                " To compute the depth, please use level='device'.",
+                UserWarning,
+            )
+        resources = _specs_qjit_intermediate_passes(qjit, original_qnode, level, *args, **kwargs)
+
+    else:
+        raise NotImplementedError(f"Unsupported level argument '{level}' for QJIT'd code.")
+
+    if isinstance(resources, dict):
+        level = list(resources.keys())
+
     return CircuitSpecs(
         resources=resources,
-        num_device_wires=len(qjit.original_function.device.wires),
-        device_name=qjit.original_function.device.name,
+        shots=original_qnode.shots,
+        device_name=device.name,
+        num_device_wires=(
+            len(original_qnode.device.wires) if original_qnode.device.wires is not None else None
+        ),
         level=level,
-        shots=qjit.original_function.shots,
     )
 
 
 def specs(
     qnode,
     level: str | int | slice = "gradient",
-    compute_depth: bool = True,
+    compute_depth: bool | None = None,
 ) -> Callable[..., CircuitSpecs]:
     r"""Provides the specifications of a quantum circuit.
 
     This transform converts a QNode into a callable that provides resource information
-    about the circuit after applying the specified amount of transforms/expansions first.
+    about the circuit after applying the specified transforms, expansions, and passes first.
 
     Args:
         qnode (.QNode | .QJIT): the QNode to calculate the specifications for.
 
     Keyword Args:
-        level (str | int | slice | iter[int]): An indication of which transforms to apply before computing the resource information.
-        compute_depth (bool): Whether to compute the depth of the circuit. If ``False``, the depth will not be included in the returned information. Default: True
+        level (str | int | slice | iter[int]): An indication of what transforms, expansions, and passes to apply before
+            computing the resource information.
+        compute_depth (bool): Whether to compute the depth of the circuit. If ``False``, circuit
+            depth will not be included in the output. Default: ``None``, meaning ``True`` where
+            depth is available, and ``False`` otherwise.
 
     Returns:
         A function that has the same argument signature as ``qnode``. This function
```

Comment:
Can't suggest, but maybe this can be added?

> A function that has the same argument signature as ``qnode``. This function
returns a :class:`~.resource.CircuitSpecs` object containing the ``qnode`` specifications, including gate types, number of wires, number of executions, and more.